libenv/sysinfo.c

```diff
@@ -3790,6 +3790,72 @@ static void SysOsVersionMajor(EvalContext *ctx)
 
 /*****************************************************************************/
 
+static void SysOsVersionMinor(EvalContext *ctx)
+{
+    // 1. parse OsRelease
+
+    DataType type_out;
+    JsonElement *os_release = (JsonElement *) EvalContextVariableGetSpecial(ctx, SPECIAL_SCOPE_SYS, "os_release", &type_out);
+
+    if (os_release == NULL)
+    {
+        goto classmatching;
+    }
+    JsonElement *version_id = JsonObjectGet(os_release, "VERSION_ID");
+    if (version_id == NULL)
+    {
+        goto classmatching;
+    }
+    const char *version_id_string = JsonPrimitiveGetAsString(version_id);
+    Item *version_tuple = SplitString(version_id_string, '.');
+    if (version_tuple->next == NULL)
+    {
+        DeleteItemList(version_tuple);
+        goto classmatching;
+    }
+    EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "os_version_minor", version_tuple->next->name, CF_DATA_TYPE_STRING, "source=agent");
+    DeleteItemList(version_tuple);
+    return;
+
+classmatching:
```

Comment:
I would try to avoid these `goto`'s if possible. You could for example wrap up everything up until the `classmatching`-label into a function. If that function returns `false`, then you proceed to look for classes matching.